server/modules/surveyRdb/repository/dataView/read.js

```diff
@@ -12,31 +12,133 @@ import * as Expression from '../../../../../core/expressionParser/expression'
+  'row_to_json(r.*) AS record',
+]
+
+const _getEditModeJoins = (viewDataNodeDef, nodeDefCols) => {
+  const { surveyId } = viewDataNodeDef
+  return [
+    // Node table; one join per column def
+    ...nodeDefCols.map((nodeDefCol, idx) => {
+      const nodeDefParentUuidColName = _getParentNodeUuidColName(viewDataNodeDef, nodeDefCol)
+      const nodeDefUuid = NodeDef.getUuid(nodeDefCol)
+      const tableNode = NodeDef.isAnalysis(nodeDefCol) ? new TableResultNode(surveyId) : new TableNode(surveyId)
+      tableNode.alias = `n${idx + 1}`
+
+      return `LEFT JOIN LATERAL ( 
+        ${tableNode.getSelect({ parentUuid: `${viewDataNodeDef.alias}.${nodeDefParentUuidColName}`, nodeDefUuid })}
+      ) AS ${tableNode.alias} ON TRUE`
+    }),
+    // Record table
+    `LEFT JOIN ${getSurveyDBSchema(surveyId)}.record r 
+    ON r.uuid = ${viewDataNodeDef.columnRecordUuid}
+  `,
+  ]
+}
+
+const _dbTransformCallbackSelect = (viewDataNodeDef, editMode, nodeDefCols) => (row) => {
+  const rowUpdated = { ...row }
+  if (editMode) {
+    // Node columns (one column for each selected node def)
+    rowUpdated.cols = {}
+    nodeDefCols.forEach((nodeDefCol) => {
+      const nodeDefColUuid = NodeDef.getUuid(nodeDefCol)
+      const nodeJson = rowUpdated[nodeDefColUuid]
+      const nodeDefParentColumnUuid = _getParentNodeUuidColName(viewDataNodeDef, nodeDefCol)
+      const parentUuid = rowUpdated[nodeDefParentColumnUuid]
+      rowUpdated.cols[nodeDefColUuid] = {
+        node: dbTransformCallbackNode(nodeJson),
+        parentUuid,
+      }
+      delete rowUpdated[nodeDefColUuid]
+    })
+    // Record column
+    rowUpdated.record = camelize(rowUpdated.record)
+    // Parent node uuid column
+    const nodeDefParentColumnUuid = _getParentNodeUuidColName(viewDataNodeDef, viewDataNodeDef.nodeDef)
+    rowUpdated.parentNodeUuid = rowUpdated[nodeDefParentColumnUuid]
+  }
+  return rowUpdated
+}
+
+/**
+ * Runs a select query on a data view associated to an entity node definition.
+ *
+ * @param {!object} params - The query parameters.
+ * @param {!Survey} [params.survey] - The survey.
+ * @param {!string} [params.nodeDefUuidTable] - The UUID of the node def associated to the view to select.
+ * @param {!Array} [params.nodeDefUuidCols] - The UUIDs of the node defs associated to the selected columns.
+ * @param {!number} [params.offset] - The query offset.
+ * @param {!number} [params.limit] - The query limit.
+ * @param {!object} [params.filterExpr] - The filter expression object.
+ * @param {Array} [params.sort=[]] - The sort conditions.
+ * @param {boolean} [params.editMode=false] - Whether to fetch row ready to be edited (fetches nodes and records).
+ * @param {boolean} [params.queryStream=false] - Whether to fetch rows to be streamed.
+ * @param {pgPromise.IDatabase} client - The database client.
+ *
+ * @returns {Promise<object>} - An object with fetched rows and selected fields.
+ */
+export const runSelect = async (params, client = db) => {
```

Comment:
this should become fetchViewData. (merge the method below)

Reply:
ok